e2e/testsuite/testsuite.go

```diff
@@ -611,6 +628,43 @@ func (s *E2ETestSuite) QueryGranterGrants(ctx context.Context, chain *cosmos.Cos
 	return grants.Grants, nil
 }
 
+// IsRelayer returns true if the provided wallet is used by a relayer.
+func (s *E2ETestSuite) IsRelayer(user ibc.Wallet) bool {
+	if s.relayers == nil {
+		return false
+	}
+	return s.relayers[user]
+}
+
+// retryNtimes retries the provided function up to the provided number of attempts.
+func (s *E2ETestSuite) retryNtimes(f func() (sdk.TxResponse, error), attempts int) (sdk.TxResponse, error) {
+	// Ignore account sequence mismatch errors.
+	allowedMessages := []string{"account sequence mismatch"}
+	var err error
+	var resp sdk.TxResponse
+	for i := 0; i < attempts; i++ {
+		// If the response's raw log doesn't contain any of the allowed prefixes we return, else, we retry.
+		resp, err = f()
+		if !containsMessage(resp.RawLog, allowedMessages) {
+			return resp, err
+		}
+		s.T().Logf("retrying tx due to ignored raw log: %s", resp.RawLog)
```

Comment:
Not sure what you mean with `ignored raw log`...

Reply:
oh `resp.RawLog` is what we look into to check if we should retry. Open to suggestions for a better log message here.

Reply:
one thing to watch out for is that sometimes the rawlog doesn't actually container a whole lot of information. It might be useful to dump the full tx so it is more clearly obvious it is an empty response. We do that [here](https://github.com/cosmos/ibc-go/blob/9bca644453430b5daab860cc6e3bb1fd3b45245d/e2e/testsuite/testsuite.go#L439)